### DISCUSSION TITLE
First impressions form a new user

### DISCUSSION THREAD

**Discussion by VKondakoff** [cablehead/stacks#73] [Ideas & Feature Requests]
Hi!

Just tried Stacks for the first time. Here are my first impressions: this is really interesting software. It is quite unusual for a first-timer, but the more you use it the more you like it. I'm pretty sure this is (potentially) really useful util.

What I like:
+1. After some initial haos-period you start to understand how things are organised and this organisation is surprisingly well thought.
+2. Stacks give you a great chance to master your command-line habits by using Pipe
+3. This is multifunctional software, not just a clipboard manager. One can use it as a diary, as a place to store drafts etc... No limits.

What I don't like:
-1. It is really hard to start using Stacks from scratch: you can not even open Stacks main window without knowing the default hotkey. Everything looks chaotic at a first glance. To be honest, I even uninstalled Stacks at one moment, deciding that I'm not ready to understand how to use it. So, there is suggestion: please, consider well-written readme for first-time users with examples of usage and easy to follow steps. Maybe some tutorial videos?
-2. Stacks and clips list is starts to scroll even before the entire hight of the screen is reached. This (again) is very confusing for the first time user, as there is always different number of stacks/clips on the screen. I'm pretty sure the scrolling should start _only_ when the entire screen is filled with stacks/clips. Video:
https://github.com/cablehead/stacks/assets/230824/de2d4280-4b0c-408e-b974-5a61a1c4e28a

What I really miss in the current Stacks state:
1. Multi-selection of stacks and clips (select several stacks/clips at once - non adjacent as well). This will allow
  1a. perform actions on several stacks/clips at once
  1b. paste several selected clips at once
2. Drag and drop clips for quick stacks/clips reorganisation. Or maybe some dedicated reorganisation-related actions?
3. Rearrange the clips in the list and paste several clip at once in their current order: for example there are four clips in the current stack: `clip1`, `clip2`, `clip3` and `clip4`. I want to paste three of them `clip1`, `clip2` and `clip4`, but in specific order (say, `clip4`, `clip1`, `clip2`). To do this I'm rearranging the clips in the list and selecting the `clip4`, `clip1`, `clip2` to paste them.

Stacks 0.15.8

**Comment by cablehead** [comment]
This is great feedback 🙏

**Comment by VKondakoff** [comment]
> 2. Drag and drop clips for quick stacks/clips reorganisation. Or maybe some dedicated reorganisation-related actions?

Aha, there is `Stash clip` action to copy clip to another stack. Great! Next logical step - to work with multiple selected clips.

**Reply by cablehead** [reply]
There's a thread on tagging / organizing clips here too: https://discord.com/channels/1182364431435436042/1242200628621606965/1242522184476262460

**Comment by VKondakoff** [comment]
> So, there is suggestion: please, consider well-written readme for first-time users with examples of usage and easy to follow steps. Maybe some tutorial videos?

Have found a basic readme on the Stacks Github homepage. Shame on me, that I was not able to find it before. I think this is because the release was located in another place, so I not checked the Github homepage carefully...

> Rearrange the clips in the list

Now I know there is a way to rearrange clips in the list using Cmd+arrows and Cmd+t.

**Reply by cablehead** [reply]
haha, thanks for persevering! Yeah, there's only patchy notes here and there, and even then lots of things aren't documented.

Once there's a good compilation of docs / tutorials, we could put them on the welcome screen: https://github.com/cablehead/stacks/issues/67 (which doesn't exist 😂)
